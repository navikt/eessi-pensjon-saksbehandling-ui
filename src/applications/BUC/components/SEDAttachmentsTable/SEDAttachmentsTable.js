import React from 'react'
import PT from 'prop-types'
import _ from 'lodash'
import { Nav, TableSorter } from 'eessi-pensjon-ui'
import './SEDAttachmentsTable.css'

const SEDAttachmentsTable = ({ attachments = {}, t }) => {
  const items = []

  Object.keys(attachments).forEach((key, index1) => {
    attachments[key].forEach((att, index2) => {
      items.push({
        key: index1 + '_' + index2,
        bucket: key,
        title: (att.tittel || att.name) +
          (att.variant ? ' + ' + att.variant.variantformat + ' (' + att.variant.filnavn + ')' : '')
      })
    })
  })

  if (_.isEmpty(items)) {
    return <Nav.Normaltekst>{t('buc:form-noAttachmentsYet')}</Nav.Normaltekst>
  }

  return (
    <TableSorter
      className='a-buc-c-sedattachmentstable'
      items={items}
      sortable={false}
      searchable={false}
      selectable={false}
      columns={[
        {
          id: 'bucket',
          label: t('ui:type'),
          type: 'string',
          renderCell: (item, value) => <Nav.EtikettLiten>{value}</Nav.EtikettLiten>
        }, {
          id: 'title', label: t('ui:title'), type: 'string'
        }, {
          id: 'buttons',
          label: '',
          type: 'object',
          renderCell: (item, value) => {
            return <div />
          }
        }
      ]}
    />
  )
}

SEDAttachmentsTable.propTypes = {
  attachments: PT.object
}

export default SEDAttachmentsTable
