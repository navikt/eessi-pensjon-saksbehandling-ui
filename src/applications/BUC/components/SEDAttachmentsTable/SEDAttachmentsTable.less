.a-buc-c-sedattachmentstable {
  td {
    padding: 0.5rem;
  }
}
