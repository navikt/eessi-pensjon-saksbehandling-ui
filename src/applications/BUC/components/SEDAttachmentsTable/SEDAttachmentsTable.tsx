import Trashcan from 'assets/icons/Trashcan'
import { HighContrastKnapp } from 'components/StyledComponents'
import { AttachedFiles, BUCAttachment } from 'declarations/buc'
import { AttachedFilesPropType } from 'declarations/buc.pt'
import { JoarkFile, JoarkFiles } from 'declarations/joark'
import _ from 'lodash'
import { EtikettLiten, Normaltekst } from 'nav-frontend-typografi'
import React from 'react'
import { useTranslation } from 'react-i18next'
import TableSorter, { Item } from 'tabell'

export interface SEDAttachmentsTableProps {
  attachments: AttachedFiles
  highContrast: boolean,
  onAttachmentsChanged ?: (files: any) => void
}

export interface SEDAttachmentsTableRow<T> extends Item {
  value: T
  namespace: string
  title: string
}

export type SEDAttachmentsTableRows = Array<SEDAttachmentsTableRow<JoarkFile | BUCAttachment>>

const SEDAttachmentsTable: React.FC<SEDAttachmentsTableProps> = ({
  attachments = {}, highContrast = false, onAttachmentsChanged
}: SEDAttachmentsTableProps): JSX.Element => {
  const items: SEDAttachmentsTableRows = []
  const { t } = useTranslation()
  Object.keys(attachments).forEach((namespace, index1) => {
    attachments[namespace].forEach((att: JoarkFile | BUCAttachment, index2: number) => {
      items.push({
        key: (att as JoarkFile).dokumentInfoId ? 'id-' + (att as JoarkFile).dokumentInfoId : 'id-' + index1 + '_' + index2,
        value: att,
        namespace: namespace,
        title: (att as JoarkFile).title || (att as BUCAttachment).name +
          ((att as JoarkFile).variant ? ' + ' +
            (att as JoarkFile).variant!.variantformat +
            ' (' + (att as JoarkFile).variant!.filnavn + ')' : '')
      })
    })
  })

  const handleDelete = (itemToDelete: SEDAttachmentsTableRow<JoarkFile>, contextAttachments: AttachedFiles) => {
    const newJoarkAttachments: JoarkFiles = _.reject(contextAttachments.joark as JoarkFiles, (att: JoarkFile) => {
      return itemToDelete.value.dokumentInfoId === att.dokumentInfoId
    })
    if (onAttachmentsChanged) {
      onAttachmentsChanged({
        ...contextAttachments,
        joark: newJoarkAttachments
      })
    }
  }

  if (_.isEmpty(items)) {
    return (
      <Normaltekst>
        {t('buc:form-noAttachmentsYet')}
      </Normaltekst>
    )
  }

  return (
    <TableSorter
      highContrast={highContrast}
      items={items}
      compact
      sortable={false}
      searchable={false}
      selectable={false}
      context={{ attachments: attachments }}
      columns={[
        {
          id: 'namespace',
          label: t('ui:type'),
          type: 'string',
          renderCell: (item: any, value: any) => <EtikettLiten>{value}</EtikettLiten>
        }, {
          id: 'title', label: t('ui:title'), type: 'string'
        }, {
          id: 'buttons',
          label: '',
          type: 'object',
          renderCell: (item: any, value: any, context: any) => (
            <>
              {item.namespace === 'joark' ? (
                <HighContrastKnapp
                  kompakt
                  mini
                  onClick={(e: any) => {
                    e.preventDefault()
                    e.stopPropagation()
                    handleDelete(item, context.attachments)
                  }}
                >
                  <Trashcan />
                </HighContrastKnapp>
              ) : <div />}
            </>
          )
        }
      ]}
    />
  )
}

SEDAttachmentsTable.propTypes = {
  attachments: AttachedFilesPropType.isRequired
}

export default SEDAttachmentsTable
