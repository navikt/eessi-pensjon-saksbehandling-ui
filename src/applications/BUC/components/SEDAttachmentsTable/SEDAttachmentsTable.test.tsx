import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import mockJoarkReduced from 'mocks/joark/joarkReduced'
import SEDAttachmentsTable, { SEDAttachmentsTableProps } from './SEDAttachmentsTable'

describe('applications/BUC/components/SEDAttachments/SEDAttachments', () => {
  let wrapper: ReactWrapper
  const initialMockProps: SEDAttachmentsTableProps = {
    attachments: {
      joark: [mockJoarkReduced[0]],
      sed: [mockJoarkReduced[2]]
    }
  }

  beforeEach(() => {
    wrapper = mount(<SEDAttachmentsTable {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure', () => {
    wrapper = mount(<SEDAttachmentsTable {...initialMockProps} />)
    expect(wrapper.exists('.a-buc-c-sedattachmentstable')).toBeTruthy()
  })
})
