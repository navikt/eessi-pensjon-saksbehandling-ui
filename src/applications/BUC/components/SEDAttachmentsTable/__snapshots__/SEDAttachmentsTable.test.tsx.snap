// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDAttachments/SEDAttachments Renders 1`] = `
<SEDAttachmentsTable
  attachments={
    Object {
      "joark": Array [
        Object {
          "datoOpprettet": 2018-12-27T13:42:24.000Z,
          "dokumentInfoId": "4",
          "journalpostId": "1",
          "tema": "foo",
          "tittel": "blue.pdf",
          "variant": Object {
            "filnavn": "23534345.pdf",
            "variantformat": "ARKIV",
          },
        },
      ],
      "sed": Array [
        Object {
          "datoOpprettet": 2018-12-17T13:42:24.000Z,
          "dokumentInfoId": "5",
          "journalpostId": "2",
          "tema": "bar",
          "tittel": "red.pdf",
          "variant": Object {
            "filnavn": "98734213487.pdf",
            "variantformat": "ARKIV",
          },
        },
      ],
    }
  }
>
  <styled.div>
    <div
      className="sc-AxmLO bXximy"
    >
      <ct
        columns={
          Array [
            Object {
              "id": "namespace",
              "label": "ui:type",
              "renderCell": [Function],
              "type": "string",
            },
            Object {
              "id": "title",
              "label": "ui:title",
              "type": "string",
            },
            Object {
              "id": "buttons",
              "label": "",
              "renderCell": [Function],
              "type": "object",
            },
          ]
        }
        items={
          Array [
            Object {
              "key": "0_0",
              "namespace": "joark",
              "title": "blue.pdf",
            },
            Object {
              "key": "1_0",
              "namespace": "sed",
              "title": "red.pdf",
            },
          ]
        }
        searchable={false}
        selectable={false}
        sortable={false}
      >
        <ThemeProvider
          theme={
            Object {
              "black": "#000000",
              "border-radius-base": "4px",
              "border-radius-large": "6px",
              "border-radius-small": "2px",
              "container-desktop": "(940px + @grid-gutter-width)",
              "container-large-desktop": "(1140px + @grid-gutter-width)",
              "container-lg": "(1140px + @grid-gutter-width)",
              "container-md": "(940px + @grid-gutter-width)",
              "container-sm": "(720px + @grid-gutter-width)",
              "container-tablet": "(720px + @grid-gutter-width)",
              "fokusFarge": "mix(@navBla,#3E3832,40%)",
              "grayBackground": "#E9E7E7",
              "grayIcon": "#B7B1A9",
              "grayInactive": "#78706A",
              "grayModia": "#333333",
              "grid-columns": "12",
              "grid-float-breakpoint": "768px",
              "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
              "grid-gutter-width": "16px",
              "line-height-large": "1.3333333",
              "line-height-small": "1.5",
              "main-background-color": "#FFFFFF",
              "main-background-other-color": "#E9E7E7",
              "main-disabled-color": "#B7B1A9",
              "main-error-color": "#BA3A26",
              "main-focus-color": "mix(@navBla,#3E3832,40%)",
              "main-font-color": "#3E3832",
              "main-hover-color": "#66CBEC",
              "main-interactive-color": "#0067C5",
              "main-success-color": "#06893A",
              "main-warning-color": "#FF9100",
              "navBla": "#0067C5",
              "navBlaDarken20": "mix(@navBla,#3E3832,80%)",
              "navBlaDarken40": "mix(@navBla,#3E3832,60%)",
              "navBlaDarken60": "mix(@navBla,#3E3832,40%)",
              "navBlaDarken80": "mix(@navBla,#3E3832,20%)",
              "navBlaLighten20": "mix(@navBla,#fff, 80%)",
              "navBlaLighten40": "mix(@navBla,#fff, 60%)",
              "navBlaLighten60": "mix(@navBla,#fff, 40%)",
              "navBlaLighten80": "mix(@navBla,#fff, 20%)",
              "navDypBla": "#005B82",
              "navDypBlaDarken20": "mix(@navDypBla,#3E3832,80%)",
              "navDypBlaDarken40": "mix(@navDypBla,#3E3832,60%)",
              "navDypBlaDarken60": "mix(@navDypBla,#3E3832,40%)",
              "navDypBlaDarken80": "mix(@navDypBla,#3E3832,20%)",
              "navDypBlaLighten20": "mix(@navDypBla,#fff, 80%)",
              "navDypBlaLighten40": "mix(@navDypBla,#fff, 60%)",
              "navDypBlaLighten60": "mix(@navDypBla,#fff, 40%)",
              "navDypBlaLighten80": "mix(@navDypBla,#fff, 20%)",
              "navGra20": "#C6C2BF",
              "navGra40": "#B7B1A9",
              "navGra60": "#78706A",
              "navGra80": "#59514B",
              "navGronn": "#06893A",
              "navGronnDarken20": "mix(@navGronn,#3E3832,80%)",
              "navGronnDarken40": "mix(@navGronn,#3E3832,60%)",
              "navGronnDarken60": "mix(@navGronn,#3E3832,40%)",
              "navGronnDarken80": "mix(@navGronn,#3E3832,20%)",
              "navGronnLighten20": "mix(@navGronn,#fff, 80%)",
              "navGronnLighten40": "mix(@navGronn,#fff, 60%)",
              "navGronnLighten60": "mix(@navGronn,#fff, 40%)",
              "navGronnLighten80": "mix(@navGronn,#fff, 20%)",
              "navLilla": "#634689",
              "navLillaDarken20": "mix(@navLilla,#3E3832,80%)",
              "navLillaDarken40": "mix(@navLilla,#3E3832,60%)",
              "navLillaDarken60": "mix(@navLilla,#3E3832,40%)",
              "navLillaDarken80": "mix(@navLilla,#3E3832,20%)",
              "navLillaLighten20": "mix(@navLilla,#fff,80%)",
              "navLillaLighten40": "mix(@navLilla,#fff,60%)",
              "navLillaLighten60": "mix(@navLilla,#fff,40%)",
              "navLillaLighten80": "mix(@navLilla,#fff,20%)",
              "navLimeGronn": "#A2AD00",
              "navLimeGronnDarken20": "mix(@navLimeGronn,#3E3832,80%)",
              "navLimeGronnDarken40": "mix(@navLimeGronn,#3E3832,60%)",
              "navLimeGronnDarken60": "mix(@navLimeGronn,#3E3832,40%)",
              "navLimeGronnDarken80": "mix(@navLimeGronn,#3E3832,20%)",
              "navLimeGronnLighten20": "mix(@navLimeGronn,#fff,80%)",
              "navLimeGronnLighten40": "mix(@navLimeGronn,#fff,60%)",
              "navLimeGronnLighten60": "mix(@navLimeGronn,#fff,40%)",
              "navLimeGronnLighten80": "mix(@navLimeGronn,#fff,20%)",
              "navLysBla": "#66CBEC",
              "navLysBlaDarken20": "mix(@navLysBla,#3E3832,80%)",
              "navLysBlaDarken40": "mix(@navLysBla,#3E3832,60%)",
              "navLysBlaDarken60": "mix(@navLysBla,#3E3832,40%)",
              "navLysBlaDarken80": "mix(@navLysBla,#3E3832,20%)",
              "navLysBlaLighten20": "mix(@navLysBla,#fff,80%)",
              "navLysBlaLighten40": "mix(@navLysBla,#fff,60%)",
              "navLysBlaLighten60": "mix(@navLysBla,#fff,40%)",
              "navLysBlaLighten80": "mix(@navLysBla,#fff,20%)",
              "navLysGra": "#E9E7E7",
              "navMorkGra": "#3E3832",
              "navOransje": "#FF9100",
              "navOransjeDarken20": "mix(@navOransje,#3E3832,80%)",
              "navOransjeDarken40": "mix(@navOransje,#3E3832,60%)",
              "navOransjeDarken60": "mix(@navOransje,#3E3832,40%)",
              "navOransjeDarken80": "mix(@navOransje,#3E3832,20%)",
              "navOransjeLighten20": "mix(@navOransje,#fff,80%)",
              "navOransjeLighten40": "mix(@navOransje,#fff,60%)",
              "navOransjeLighten60": "mix(@navOransje,#fff,40%)",
              "navOransjeLighten80": "mix(@navOransje,#fff,20%)",
              "navRod": "#C30000",
              "navRodDarken20": "mix(@navRod,#3E3832,80%)",
              "navRodDarken40": "mix(@navRod,#3E3832,60%)",
              "navRodDarken60": "mix(@navRod,#3E3832,40%)",
              "navRodDarken80": "mix(@navRod,#3E3832,20%)",
              "navRodLighten20": "mix(@navRod,#fff,80%)",
              "navRodLighten40": "mix(@navRod,#fff,60%)",
              "navRodLighten60": "mix(@navRod,#fff,40%)",
              "navRodLighten80": "mix(@navRod,#fff,20%)",
              "orangeFocus": "#FFBD66",
              "orangeFocusDarken20": "mix(@orangeFocus,#3E3832,80%)",
              "orangeFocusDarken40": "mix(@orangeFocus,#3E3832,60%)",
              "orangeFocusDarken60": "mix(@orangeFocus,#3E3832,40%)",
              "orangeFocusDarken80": "mix(@orangeFocus,#3E3832,20%)",
              "orangeFocusLighten20": "mix(@orangeFocus,#fff,80%)",
              "orangeFocusLighten40": "mix(@orangeFocus,#fff,60%)",
              "orangeFocusLighten60": "mix(@orangeFocus,#fff,40%)",
              "orangeFocusLighten80": "mix(@orangeFocus,#fff,20%)",
              "padding-base-horizontal": "12px",
              "padding-base-vertical": "6px",
              "padding-large-horizontal": "16px",
              "padding-large-vertical": "10px",
              "padding-small-horizontal": "10px",
              "padding-small-vertical": "5px",
              "padding-xs-horizontal": "5px",
              "padding-xs-vertical": "1px",
              "pension-orange": "rgb(255, 167, 51)",
              "pension-orange-alpha": "rgba(255, 167, 51, 0.5)",
              "pension-orange-alpha-80": "rgba(255, 167, 51, 0.8)",
              "pinkErrorBg": "#F3E3E3",
              "redError": "#BA3A26",
              "redErrorDarken20": "mix(@redError,#3E3832,80%)",
              "redErrorDarken40": "mix(@redError,#3E3832,60%)",
              "redErrorDarken60": "mix(@redError,#3E3832,40%)",
              "redErrorDarken80": "mix(@redError,#3E3832,20%)",
              "redErrorLighten20": "mix(@redError,#fff,80%)",
              "redErrorLighten40": "mix(@redError,#fff,60%)",
              "redErrorLighten60": "mix(@redError,#fff,40%)",
              "redErrorLighten80": "mix(@redError,#fff,20%)",
              "screen-desktop": "992px",
              "screen-lg": "1200px",
              "screen-lg-desktop": "1200px",
              "screen-lg-min": "1200px",
              "screen-md": "992px",
              "screen-md-max": "(@screen-lg-min - 1)",
              "screen-md-min": "992px",
              "screen-sm": "768px",
              "screen-sm-max": "(@screen-md-min - 1)",
              "screen-sm-min": "768px",
              "screen-tablet": "768px",
              "screen-xs-max": "(@screen-sm-min - 1)",
              "type": "theme",
              "white": "#FFFFFF",
            }
          }
        >
          <styled.div
            className="tabell"
          >
            <div
              className="sc-AxiKw dnnAxe tabell"
            >
              <styled.div>
                <div
                  className="sc-AxhCb jZxptO"
                >
                  <styled.table
                    cellSpacing="0"
                    className="c-tableSorter__table"
                  >
                    <table
                      cellSpacing="0"
                      className="sc-AxgMl gcDAfa c-tableSorter__table"
                    >
                      <thead>
                        <tr
                          className="c-tableSorter__header"
                        >
                          <th
                            style={
                              Object {
                                "width": 1,
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "alignItems": "center",
                                  "display": "flex",
                                }
                              }
                            />
                          </th>
                          <th
                            aria-sort="none"
                            className="header "
                            key="namespace"
                            role="columnheader"
                          >
                            ui:type
                          </th>
                          <th
                            aria-sort="none"
                            className="header "
                            key="title"
                            role="columnheader"
                          >
                            ui:title
                          </th>
                          <th
                            aria-sort="none"
                            className="header"
                            key="buttons"
                            role="columnheader"
                          />
                        </tr>
                      </thead>
                      <tbody>
                        <tr
                          aria-selected={false}
                          className="slideAnimate"
                          key="0_0"
                          style={
                            Object {
                              "animationDelay": "0s",
                            }
                          }
                        >
                          <td />
                          <td
                            className=""
                            key="0"
                          >
                            <EtikettLiten>
                              <TypografiBase
                                type="etikettLiten"
                              >
                                <p
                                  className="typo-etikett-liten"
                                >
                                  joark
                                </p>
                              </TypografiBase>
                            </EtikettLiten>
                          </td>
                          <td
                            className=""
                            key="1"
                          >
                            <n>
                              <t
                                type="normaltekst"
                              >
                                <p
                                  className="typo-normal"
                                >
                                  <span>
                                    blue.pdf
                                  </span>
                                </p>
                              </t>
                            </n>
                          </td>
                          <td
                            className=""
                            key="2"
                          >
                            <div />
                          </td>
                        </tr>
                        <tr
                          aria-selected={false}
                          className="slideAnimate"
                          key="1_0"
                          style={
                            Object {
                              "animationDelay": "0.04s",
                            }
                          }
                        >
                          <td />
                          <td
                            className=""
                            key="0"
                          >
                            <EtikettLiten>
                              <TypografiBase
                                type="etikettLiten"
                              >
                                <p
                                  className="typo-etikett-liten"
                                >
                                  sed
                                </p>
                              </TypografiBase>
                            </EtikettLiten>
                          </td>
                          <td
                            className=""
                            key="1"
                          >
                            <n>
                              <t
                                type="normaltekst"
                              >
                                <p
                                  className="typo-normal"
                                >
                                  <span>
                                    red.pdf
                                  </span>
                                </p>
                              </t>
                            </n>
                          </td>
                          <td
                            className=""
                            key="2"
                          >
                            <div />
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </styled.table>
                  <styled.div>
                    <div
                      className="sc-AxheI fzKHgI"
                    >
                      <y
                        highContrast={false}
                        initialPage={1}
                        itemsPerPage={10}
                        numberOfItems={2}
                        onChange={[Function]}
                      />
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </ThemeProvider>
      </ct>
    </div>
  </styled.div>
</SEDAttachmentsTable>
`;
